packages/backend/src/apps/m365-excel/dynamic-data/list-worksheets/index.ts

```diff
@@ -19,6 +20,14 @@ const dynamicData: IDynamicData = {
     const authData = getRegisteredAuthData($)
     const tenant = getM365TenantInfo(authData.tenantKey)
 
+    // Did not want to open a workbook session as user could just be casually
+    // browsing through files.
+    // FIXME (ogp-weeloong): move to a central file metadata cache to remove
+    // need for this check
+    if (await isFileTooSensitive(tenant, fileId as string, $.http)) {
+      throw new Error(`File is too sensitive!`)
+    }
```

Comment:
think u could put the throw statement in the function and rename it to `validateFileSensitivity`

Reply:
agreed. Will do this in a later PR because I refactor this file quite heavily in my cache PR.